### ISSUE TITLE
SCSS - Parser breaks on lines with 2 string interpolations

### ISSUE THREAD

**Issue by colin-marshall** [sasstools/sass-lint#644]
Example:
https://github.com/zurb/foundation-sites/blob/develop/scss/grid/_classes.scss#L85
https://github.com/zurb/foundation-sites/blob/develop/scss/grid/_flex-grid.scss#L195

``` scss
.#{$-zf-size}-#{$i} { }
```

Here's a slightly different string interpolation that also throws a fatal error:
https://github.com/zurb/foundation-sites/blob/develop/scss/grid/_layout.scss#L25

``` scss
&:nth-of-type(#{$n}n+1) { }
```

Interestingly, the 3 interpolated strings in one line here don't break the parser:
https://github.com/zurb/foundation-sites/blob/develop/scss/grid/_classes.scss#L70

``` scss
.#{$column}.#{$row}.#{$row} { }
```

**Comment by bgriffith** [comment]
There goes the hope that we'd knocked all those interpolation parse errors on the head! Thanks for the report :+1:

**Comment by egaudette** [comment]
I'm also seeing this when linting SCSS with the 'stylish' formatter. If you believe that there is any additional information that I may be able to provide, please don't hesitate to request it.

Thanks.

LN# 259:
&:nth-of-type(#{$n}n+1) { }

        [exec] [11:47:47] Finished 'sass' after 3.27 s
        [exec] _responsive_grid_functions.scss
        [exec]   259:1  error  Please check validity of the block starting from line #259  Fatal
        [exec] ✖ 1 problem (1 error, 0 warnings)
        [exec] [11:47:48] 'sass-lint' errored after 3.98 s
        [exec] [11:47:48] Error in plugin 'sass-lint'

**Comment by nmcclay** [comment]
I also encountered this issue after updating today, is there any work around for this issue?
```
#{$btn-namespace}-#{$btn-name} {
    @include button($btn-color-set);
}
```

**Comment by scottyeck** [comment]
FWIW, as per @colin-marshall 's second example, I've encountered it as well.

```sass
&:nth-of-type(#{$per-row}n+1) {
	clear: both;
}
```

No error is thrown if the string-concatenation is performed in the interpolated block.

```sass
&:nth-of-type(#{$per-row + "n+1"}) {
	clear: both;
}
```

**Comment by colin-marshall** [comment]
@scottyeck good find! So would using the syntax that makes it break be considered "bad" SCSS? Or would it not compile if that was the case?

**Comment by scottyeck** [comment]
From a readability perspective, I'd say the first of my two options is preferable, since it's more obvious that the interpolation is part of the construction of an _n_th of type formula. The pickiest of picky, I'm aware.

That said, my guess thus far is that the parser is breaking on the `+` token, as it's interpreting it as an operator when, in this case, we just want regular old output. I'm going to file a bug with [gonzales-pe](https://github.com/tonyganch/gonzales-pe) and try to track it down.

**Comment by bgriffith** [comment]
Just to let you guys know, I'm working on a fix for this. Seem to have fixed the the hyphen issue and now looking at the pseudo class issue. If you have any more examples I'd really appreciate if you could post them so I can make sure the testing is thorough :+1:

**Comment by scottyeck** [comment]
@bgriffith Thanks. I'm happy to lend a hand if necessary as well. Lmk. 👌

**Comment by bgriffith** [comment]
Thanks for the offer @scottyeck ! Only just come up for air... so I've fixed the AST output and have tests for the following pseudo combos:

```css
:nth-child(#{$i})
:nth-child(#{$i}n)
:nth-child(#{$i}n+1)
:nth-child(#{$i}n-1)
:nth-child(-#{$i}n)
:nth-child(-#{$i}n+1)
:nth-child(-#{$i}n+#{$j})
:nth-child(-#{$i}n-1)
:nth-child(-#{$i}n-#{$j})
```

Hopefully be able to get a patch out to gonzales in next few days.

**Comment by omgaz** [comment]
Same issue for this example:

```scss
.square-component {
  @mixin size($diameter) {
    &-#{$diameter} {
      height: $diameter + px;
      width: $diameter + px;
    }
  }

  @include size(70);
}
```

**Comment by colin-marshall** [comment]
@bgriffith any update on the status of this? Thanks!!!

**Comment by bgriffith** [comment]
@colin-marshall So I think it's all fixed 🌟! I've run the latest build of Sasslint (develop branch) and linted the `foundation-sites` scss folder with no parse issues. Just looking to merge a few more PR's before we release a new version.

I also plan on checking and closing all the related issues here and on the gonzales repo just to make doubly sure I've not missed anything :+1:

**Comment by bgriffith** [comment]
@colin-marshall So in short - stay tuned haha

**Comment by DanPurdy** [comment]
So all of the examples in this issue are now parsing correctly in sass-lint 1.9.0 except for the issue highlighted by @nmcclay

**Comment by colin-marshall** [comment]
Thanks guys! I'll test it out this week.

**Comment by abpetkov** [comment]
@DanPurdy @bgriffith Came across the same fatal error with the code below. Running version 1.10.2

```
.#{$spacing}-#{$orientation}-#{$type} { #{$spacing}-#{$orientation}: $size !important; }
```

Looks like the problem is with the property rule (BTW this is usually something like `padding-left`, `margin-right`, etc.). It doesn't error out if it's a straightforward property.

Any workaround?

**Comment by DanPurdy** [comment]
@abpetkov It's a fatal error with our AST [gonzales-pe](https://github.com/tonyganch/gonzales-pe/issues) If you could file an issue there that would be really helpful.

It's the second piece of interpolation in your property that is causing it.

```scss
.#{$spacing}-#{$orientation}-#{$type} {
  #{$spacing}: $size !important;
}
```

The above parses fine for example

**Comment by abpetkov** [comment]
@DanPurdy Right, confirming that the property interpolation causes it. Will open an issue about that. Thanks!

**Comment by ytilis** [comment]
Any update on the status of property interpolation?

I run into it in the following line:

    border-#{opposite($d-y)}-#{opposite($d-x)}-radius: 0;

Which forces me to use this uglier line as a workaround:

    border-#{opposite($d-y) + '-' + opposite($d-x)}-radius: 0;

**Comment by bgriffith** [comment]
This is fixed in the newly released version v1.11.0. Please let me know if you still have issues. Thanks!
